forecasters/armadillo/R/armadillo_forecaster_optimizer.R

```diff
@@ -0,0 +1,70 @@
+#' Fit Larry and Valerie's mobility model
+#'
+#' @param L length of observed response series.
+#' @param At normalized parameter.
+#' @param beta mobility function parameter.
+#' @param alpha mobility function parameter.
+#' @param mu mobility function parameter.
+#' @param sigma mobility function parameter.
+#' @param M mobility series.
+#' @param DC death delay curve.
+#' @param before_pan if TRUE, the first estimated response must be 0.
+#'
+#' @return estmiated response series.
+#' @importFrom stats pnorm
+#' @importFrom optimization optim_sa
+#'
+Mean.fun <- function(L, At, beta, alpha, mu, sigma, M, DC, before_pan) {
+  Beta <- beta + alpha * stats::pnorm(M[1:L], mu, sigma)
+  BetaSum <- cumsum(c(0, Beta))
+  BetaSum <- BetaSum[-(L + 1)]
+  if (before_pan) {
+    out <- rep(0, L)
+  } else {
+    out <- rep(1, L)
+  }
+  DC0 <- DC
+  for (d in 1:L) {
+    DC0 <- c(0, DC0[-L])
+    out <- out + DC0 * exp(BetaSum[d])
+  }
+  return(At * out)
```

Comment:
Here and elsewhere, better practice is to use `At * out` alone rather than explicitly calling `return()`. Only need to use `return()` if it happens in some condition in the middle of the function.

Reply:
Thanks! Change all the usage of `return()` and rename the loss function.